spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/SparkWrite.java

```diff
@@ -117,6 +124,18 @@
     this.dsSchema = dsSchema;
     this.extraSnapshotMetadata = writeConf.extraSnapshotMetadata();
     this.partitionedFanoutEnabled = writeConf.fanoutWriterEnabled();
+    this.requiredDistribution = requiredDistribution;
+    this.requiredOrdering = requiredOrdering;
+  }
+
+  @Override
```

Comment:
I don't let users to request a fixed number of shuffle partitions. I hope AQE would be smart enough and will make a better choice.

Reply:
Unfortunately, Spark won't use AQE here due to SPARK-34230. It will, eventually. That's why I still keep that logic.